### ISSUE TITLE
hinthash is not saved during compilation time

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#220]
This is probably not a big deal, as this is mainly an internal variable

Env:
perl 5.20.2
B::C using bc522 with HEAD @734e776 ( cannot use 23016e1 with 5.20 )

````
> cat op/caller.t
#!./perl
# Tests for caller()

BEGIN { $^H{dooot} = -42 }

sub hint_fetch {
    my $key = shift;

    my @results = caller(0);
    $results[10]->{$key};
}

print qq{ok\n} if hint_fetch('dooot') == -42;
> perl op/caller.t
ok
> perlcc -r op/caller.t
````

**Comment by rurban** [comment]
That's the compile-time PL_hints hash. 
The run-time cop_hints are stored correctly, but here the bug is that the run-time cop_hints (from main_cv) should take over the values from PL_hints.

**Comment by rurban** [comment]
use open #31 is related to not storing %^H

**Comment by rurban** [comment]
testcases: t/testc.sh -q -k -O0 130 180 203 2900 220 250 302 304

**Comment by rurban** [comment]
Fixed with:
commit 2855398bfc151877cf4132a38263a14ace64fb5e
Author: Reini Urban <rurban@cpanel.net>
Date:   Fri Nov 20 09:31:00 2015 +0100

    C: fix labels with hints_hash

    store labels after the hints_hash, so that they
    are not overwritten, just added.
    Also skip labels (key=":") in the hints_hash loop.
    Fixes remaining fails 240,242

commit 0da3356e1c661a983ee497bf8d051ab32652265f
Author: Reini Urban <rurban@cpanel.net>
Date:   Fri Nov 20 00:19:28 2015 +0100

    C: support COPHH from 5.10 - 5.12 also

    < 5.10 %^H is unsupported. no hints hash,
    just lexical cop_io defaults (NYI)

commit 0c3697d03033fb9548f591d17266cf70ea1fcf6f
Author: Reini Urban <rurban@cpanel.net>
Date:   Thu Nov 19 23:51:17 2015 +0100

    C: cache cophh ptrs, and support utf8 %^H keys

commit cab48ce67ce0ebd30114c361941c1c8c99c5e2d3
Author: Reini Urban <rurban@cpanel.net>
Date:   Thu Nov 19 23:18:47 2015 +0100

    C: delete dead global hints_hash code

    hints_hash is only accessible from COPs

commit 4ab03cca6779a7a46ed7a68a67b133ae41864ea1
Author: Reini Urban <rurban@cpanel.net>
Date:   Thu Nov 19 14:44:45 2015 +0100

    C 1.52_22: save COP->hints_hash

    Fixes GH #220

    move global state to save_context
    This has nothing to do in modules.

    But the cophh/B::RHE chain is not yet stored efficiently.

    Fixes t/testc.sh -q -k -O0 130 180 220 250 302
    but broke 240 242
    use open #31 not yet, i.e. $^H{open<}
